src/registry.ts

```diff
@@ -0,0 +1,67 @@
+import { promises as fs } from 'fs';
```

Comment:
Can you add some comments in this file to link to the original implementation in setup-node? (so we can keep track of changes if they have to happen)